core/Settings/Setting.php

```diff
@@ -201,7 +201,17 @@ public function save()
      */
     public function getValue()
     {
-        return $this->storage->getValue($this->name, $this->defaultValue, $this->type);
+        $value = $this->storage->getValue($this->name, $this->defaultValue, $this->type);
+
+        $config = $this->configureField();
```

Comment:
A while ago we refactored the entire settings API so we basically wouldn't need to `configureField()` when getting the value as it is a performance issue. Neither should we `validateValue` ideally. Any chance we can tweak the UI component to return the value in an alternative format through a flag or something? 
We'd really need to make sure to leave the `getValue` as it is... I probably should have added a comment about this...